tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/stack_trace/stack_trace_component.ts

```diff
@@ -59,5 +75,31 @@ export class StackTraceComponent {
     lineno: number;
   }>();
 
+  @ViewChild('stackFrameArray')
+  private readonly stackFrameArray!: ElementRef<HTMLDivElement>;
+
   CodeLocationType = CodeLocationType;
+
+  ngAfterViewChecked(): void {
+    if (this.stackFrameArray === undefined) {
+      return;
+    }
+    const stackElement = this.stackFrameArray.nativeElement;
+    const focusedFrameElement: HTMLElement | null = stackElement.querySelector(
+      `.focused-stack-frame`
+    );
+    if (focusedFrameElement !== null) {
+      // Scroll the focused frame into view when there is a focused frame.
+      focusedFrameElement.scrollIntoView({block: 'nearest'});
+      return;
+    }
+    const lastFrameElement: HTMLElement | null = stackElement.querySelector(
+      '.stack-frame-container:last-child'
+    );
+    if (lastFrameElement) {
+      // Scroll the last frame into view only when there is no stack frame being
+      // focused on.
+      lastFrameElement.scrollIntoView({block: 'nearest'});
+    }
```

Comment:
If you prefer, L91-L103 might be simplified by assuming the last element is always at the bottom.

```
if (focusedFrameElement) {
  focusedFrameElement.scrollIntoView({block: 'nearest'});
} else {
  stackElement.scrollTop = stackElement.scrollHeight;
}
```